diff --git a/kernel/gen_kheaders.sh b/kernel/gen_kheaders.sh
index 8ffa6b894ef0..7a0ae4bf9113 100755
--- a/kernel/gen_kheaders.sh
+++ b/kernel/gen_kheaders.sh
@@ -1,6 +1,12 @@
 #!/bin/bash
 # SPDX-License-Identifier: GPL-2.0
 
+# Check if cpio avaible
+if ! hash cpio 2>/dev/null; then
+echo "cpio not installed on environment, exit 127."
+exit 127
+fi
+
 # This script generates an archive consisting of kernel headers
 # for CONFIG_IKHEADERS.
 set -e
@@ -8,7 +14,7 @@ sfile="$(readlink -f "$0")"
 outdir="$(pwd)"
 tarfile=$1
 cpio_dir=$outdir/$tarfile.tmp
-cpio=$KBUILD_SRC/tools/build/cpio
+
 dir_list="
 include/
 arch/$SRCARCH/include/
@@ -58,14 +64,14 @@ mkdir $cpio_dir
 pushd $srctree > /dev/null
 for f in $dir_list;
 	do find "$f" -name "*.h";
-done | $cpio --quiet -pd $cpio_dir
+done | cpio --quiet -pd $cpio_dir
 popd > /dev/null
 
 # The second CPIO can complain if files already exist which can
 # happen with out of tree builds. Just silence CPIO for now.
 for f in $dir_list;
 	do find "$f" -name "*.h";
-done | $cpio --quiet -pd $cpio_dir >/dev/null 2>&1
+done | cpio --quiet -pd $cpio_dir >/dev/null 2>&1
 
 # Remove comments except SDPX lines
 find $cpio_dir -type f -print0 |
