diff --git a/Makefile b/Makefile
index cd9ff1336..c5776802d 100644
--- a/Makefile
+++ b/Makefile
@@ -425,6 +425,10 @@ LINUXINCLUDE    := \
 		$(if $(KBUILD_SRC), -I$(srctree)/include) \
 		-I$(srctree)/drivers/misc/mediatek/include \
 		-I$(objtree)/include \
+		-I$(srctree)/drivers/misc/mediatek/include/mt6768 \
+		-I$(objtree)/include \
+		-I$(srctree)/drivers/misc/mediatek/include/mt-plat/mt6768/include/mach \
+		-I$(objtree)/include/ \
 		$(USERINCLUDE)
 
 KBUILD_AFLAGS   := -D__ASSEMBLY__
diff --git a/drivers/devfreq/helio-dvfsrc-mt6768.h b/drivers/devfreq/helio-dvfsrc-mt6768.h
index ec91caa43..d5492ad74 100644
--- a/drivers/devfreq/helio-dvfsrc-mt6768.h
+++ b/drivers/devfreq/helio-dvfsrc-mt6768.h
@@ -14,7 +14,7 @@
 #ifndef __HELIO_DVFSRC_MT6768_H
 #define __HELIO_DVFSRC_MT6768_H
 
-#include <mach/upmu_hw.h>
+#include <upmu_hw.h>
 
 #define PMIC_VCORE_ADDR		PMIC_RG_BUCK_VCORE_VOSEL
